retroshare-gui/src/gui/gxschannels/CreateGxsChannelMsg.cpp

```diff
@@ -449,11 +450,10 @@ void CreateGxsChannelMsg::addAttachment(const std::string &path)
 	}
 
 	FileInfo fInfo;
-	std::string filename;
+	std::string filename = RsDirUtil::getTopDir(path);
```

Comment:
getTopDir() returns the file name??? Either a typo, or the name of the function is really not well chosen.

Reply:
At first glance it seems to me that that function just split the path by / and return the last piece, so in case of a file path it is a file name, in case it is a dir path is the last dir name, but i agree that the function name is at least misleading...